clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -2241,19 +2244,23 @@ public DescribeConsumerGroupsResult describeConsumerGroups(final Collection<Stri
         for (String groupId: groupIds) {
             if (groupIdIsUnrepresentable(groupId)) {
                 KafkaFutureImpl<ConsumerGroupDescription> future = new KafkaFutureImpl<>();
-                future.completeExceptionally(new ApiException("The given group id '" +
+                future.completeExceptionally(new InvalidGroupIdException("The given group id '" +
                         groupId + "' cannot be represented in a request."));
                 futures.put(groupId, future);
             } else if (!futures.containsKey(groupId)) {
                 futures.put(groupId, new KafkaFutureImpl<ConsumerGroupDescription>());
             }
         }
 
-        // TODO: we should consider grouping the request per coordinator and send one request with a list of
+        // TODO: KAFKA-6788, we should consider grouping the request per coordinator and send one request with a list of
         // all consumer groups this coordinator host
         for (final String groupId : groupIds) {
-            final long nowFindCoordinator = time.milliseconds();
-            final long deadline = calcDeadlineMs(nowFindCoordinator, options.timeoutMs());
+            // skip sending request for those futures that already failed.
+            if (futures.get(groupId).isCompletedExceptionally())
```

Comment:
nit: couldn't we be iterating over the `futures` map?

Reply:
ack.